File: runinator-command-center/src/core/utils/__tests__/secrets.test.ts
import { describe, expect, it } from "vitest";
import { parseSecretRef, secretRef, secretRefLabel } from "../secrets";

describe("secret utils", () => {
  it("round trips secret references", () => {
    const value = secretRef("github", "token/main");
    expect(value).toBe("secret://github/token%2Fmain");
    expect(parseSecretRef(value)).toEqual({ scope: "github", name: "token/main" });
    expect(secretRefLabel(value)).toBe("github/token/main");
  });

  it("ignores non-secret values", () => {
    expect(parseSecretRef("plain text")).toBeNull();
    expect(secretRefLabel("plain text")).toBe("");
  });
});

File: runinator-command-center-flutter/test/core/utils/secrets_test.dart
import 'package:runinator_command_center_flutter/core/utils/secrets.dart';
import 'package:test/test.dart';

void main() {
  group('secret utils', () {
    test('round trips secret references', () {
      const value = 'secret://github/token%2Fmain';
      expect(secretRef('github', 'token/main'), value);
      expect(parseSecretRef(value)?.scope, 'github');
      expect(parseSecretRef(value)?.name, 'token/main');
      expect(secretRefLabel(value), 'github/token/main');
    });

    test('ignores non-secret values', () {
      expect(parseSecretRef('plain text'), isNull);
      expect(secretRefLabel('plain text'), '');
    });
  });
}
